### ISSUE TITLE
Issue processing a particular JSON file

### ISSUE THREAD

**Issue by ashleydavis** [philbooth/bfj#57]
I'm having trouble parsing a particular JSON file.

Example repo is here: https://github.com/JavaScript-Data-Wrangling/Chapter-7

This is the code file that has the problem: 

https://github.com/JavaScript-Data-Wrangling/Chapter-7/blob/master/listing-2.js

Run it like this:

```
cd Chapter-7
node listing-2.js
```

I get this error:

```
> node listing-2.js
Unhandled rejection Error: JSON error: encountered `EOF` at line 11, column 4 where `}` was expected.
    at Object.create (C:\Dropbox\Projects\JSDW\Github\Chapter-7\node_modules\bfj\src\error.js:6:17)
    at fail (C:\Dropbox\Projects\JSDW\Github\Chapter-7\node_modules\bfj\src\walk.js:317:13)
    at endWalk (C:\Dropbox\Projects\JSDW\Github\Chapter-7\node_modules\bfj\src\walk.js:604:7)
    at tryCatcher (C:\Dropbox\Projects\JSDW\Github\Chapter-7\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\Dropbox\Projects\JSDW\Github\Chapter-7\node_modules\bluebird\js\release\promise.js:512:31)
    at Promise._settlePromise (C:\Dropbox\Projects\JSDW\Github\Chapter-
```

This is the JSON file: https://github.com/JavaScript-Data-Wrangling/Chapter-7/blob/master/data/weather-stations.json

I've double checked the JSON file in an online JSON validator so I think it's ok.

**Comment by philbooth** [comment]
There's a couple of problems here, one in your script and one in bfj.

On line 85 of your script, `curObject` is undefined. It's defined inside the `openJsonInputStream` closure but that's not in scope here.

That then throws an error back in to bfj via the `bfj.events.endObject` event handler on line 49 and chaos ensues. The `walk` API does not expect any errors to be thrown and the upshot is that very misleading error message you saw about the JSON being invalid.

I'll fix bfj to sanely report errors that come back from event handlers. I've pretty much treated the `walk` and `eventify` APIs as internal up to now, so haven't paid much attention to making them robust. It's good to see them getting exercised externally.

In the meantime, if you fix the problem in your script you should find that it parses the file fine. You might want to consider including a linter in your workflow for the book as well! 😁

**Comment by philbooth** [comment]
I fixed the bfj part of this in version `5.1.1`. Feel free to re-open if you still see any problems!

**Comment by ashleydavis** [comment]
Thanks!

curObject was defined by the time line 85 is executed.

It all works after your fix! I appreciate your effort on this.
